virtual_rainforest/core/config.py

```diff
@@ -421,32 +421,35 @@ def build_config(self) -> None:
     def build_schema(self) -> None:
         """Build a schema to validate the model configuration.
 
-        This method identifies the modules to be configured from the config
-        `core.modules` entry, using the core schema default if this entry is missing.
-        This sets the requested modules to be used in the configured model. The schemas
-        for the requested modules are then loaded and combined using the
-        :meth:`~virtual_rainforest.core.schema.merge_schemas` function to generate a
-        single validation schema for model configuration.
+        This method identifies the modules to be configured from the top-level
+        configuration keys, setting the requested modules to be used in the configured
+        simulation. The schemas for the requested modules are then loaded and combined
+        using the :meth:`~virtual_rainforest.core.schema.merge_schemas` function to
+        generate a single validation schema for model configuration.
         """
 
-        # Register the core module components and then extract the modules requested in
-        # the configuration, falling back to the schema defaults.
+        # Extract the requested modules, which are the top-level config keys.
+        requested_modules: list[str] = list(self.keys())
+
+        # Must include core
+        if "core" not in requested_modules:
```

Comment:
As `core` must always be there, why raising an error if not present? Just use it with the default options in that case.

Reply:
Yeah - my thinking here was that the simplest possible TOML config lists all of the modules wanted, so for consistency should include `[core]` so that users have to keep that configuration element obvious. All the sections are there to be expanded right from the most basic permitted config.

As a user though, I'd definitely be irritated if my config failed on that step.

Reply:
I see the beauty of that symmetry, but asking the user to include it when it is not optional and might be used with the default configuration, seems, as you say, annoying.

Reply:
Bah. Form over function,  I say 😄

Reply:
OK - reverted to `[core]` being optional, and logging a message to that effect.